toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt

```diff
@@ -52,30 +62,121 @@ public class MapViewScope(internal val mapView: MapView)
+    location: Point,
+    modifier: Modifier = Modifier,
+    offset: Offset = Offset.Zero,
+    rotateOffsetWithGeoView: Boolean = false,
+    content: @Composable BoxScope.() -> Unit
+) {
 
     if (mapView.map?.loadStatus?.collectAsState()?.value == LoadStatus.Loaded) {
-        val calloutScreenCoordinate: ScreenCoordinate = mapView.locationToScreen(location)
-        Box(
-            modifier = Modifier
-                .graphicsLayer {
-                    translationX = calloutScreenCoordinate.x.toFloat()
-                    translationY = calloutScreenCoordinate.y.toFloat()
-                }
-                .wrapContentSize()
-                .background(Color.White)
-                .border(
-                    border = BorderStroke(2.dp, Color.LightGray),
-                    shape = MaterialTheme.shapes.medium
-                )
+
+        val leaderScreenCoordinate = getLeaderScreenCoordinate(
+            mapView,
+            location,
+            offset,
+            rotateOffsetWithGeoView,
         )
-        {
-            this.content()
+        leaderScreenCoordinate?.let {
+            Box(
+                modifier = Modifier
+                    .graphicsLayer {
+                        translationX = leaderScreenCoordinate.x.toFloat()
+                        translationY = leaderScreenCoordinate.y.toFloat()
+                    }
+                    .wrapContentSize()
+                    .background(Color.White)
+                    .border(
+                        border = BorderStroke(2.dp, Color.LightGray),
+                        shape = MaterialTheme.shapes.medium
+                    )
+            )
+            {
+                this.content()
+            }
+        }
+    }
+}
+
+/**
+ * Returns the ScreenCoordinate for the location [Point] on [GeoView].
+ *
+ * @param geoView the GeoView
+ * @param location the location in geographical coordinates
+ * @param offset the offset in screen coordinates from the geographical location at which to place the callout
+ * @param rotateOffsetWithGeoView specifies whether the screen offset is rotated with the [GeoView]. The Screen offset
+ *       will be rotated with the [GeoView] when true, false otherwise.
+ * @return A [ScreenCoordinate] for the screen in pixels or null if the location is not visible
+ * @since 200.5.0
+ */
+private fun getLeaderScreenCoordinate(
+    geoView: GeoView,
+    location: Point,
+    offset: Offset,
+    rotateOffsetWithGeoView: Boolean
+): ScreenCoordinate? {
+    val geoViewRotation = geoView.rotation()
+    val locationToScreen = if (geoView is MapView) {
+        geoView.locationToScreen(location)
+    } else {
+        // geoView is a SceneView
+        val locationToScreenResult = (geoView as SceneView).locationToScreen(location)
+        if (locationToScreenResult?.visibility == SceneLocationVisibility.Visible) {
+            return locationToScreenResult.screenPoint
+        }
+        null
+    }
```

Comment:
This could use a when statement instead: 
```suggestion
val locationToScreen = when (geoView) {
        is MapView -> geoView.locationToScreen(location)
        is SceneView -> {
            val locationToScreenResult = geoView.locationToScreen(location)
            if (locationToScreenResult?.visibility == SceneLocationVisibility.Visible) {
                return locationToScreenResult.screenPoint
            }
            null
        }
    }
```